spec/forms/waste_exemptions_engine/check_site_address_form_spec.rb

```diff
@@ -29,20 +29,46 @@ module WasteExemptionsEngine
       end
     end
 
+    let(:address_attributes) { (TransientAddress.attribute_names - %w[id created_at updated_at address_type]) }
+
     describe "submit" do
       let(:form) { build(:check_site_address_form) }
 
       subject do
         form.submit(temp_reuse_address_for_site_location: temp_reuse_address_for_site_location)
       end
 
-      context "when temp_reuse_address_for_site_location is true for the operators address" do
+      context "when temp_reuse_address_for_site_location is wet to operator_address_option" do
         let(:temp_reuse_address_for_site_location) { "operator_address_option" }
 
-        it "assigns the operator address as the site address" do
+        it "uses the operator address as the site address" do
+          subject
+
+          address_attributes.each do |attr|
+            expect(form.transient_registration.site_address.send(attr)).to eq(form.operator_address.send(attr))
+          end
+        end
+      end
+
+      context "when temp_reuse_address_for_site_location set to contact_address_option" do
+        let(:temp_reuse_address_for_site_location) { "contact_address_option" }
+
+        it "uses the contact address as the site address" do
+          subject
+
+          address_attributes.each do |attr|
+            expect(form.transient_registration.site_address.send(attr)).to eq(form.operator_address.send(attr))
```

Comment:
Should compare site_address with contact_address, not with operator_address. Maybe vary the operator or contact address slightly to avoid having identical comparison targets.

Reply:
Updated the factory to create different addresses and changed this spec to match the Contact address